### PR TITLE
Type check error handling (second try)

### PR DESCRIPTION
**Pull request by jdidion**
* Add an optional errorHandler function parameter to TypeInfer. TypeInfer delegates all errors to this handler if defined, otherwise throws TypeException as before.
* Add an optional errorHandler function parameter to ParseAll classes.
* Add TextSource parameters to all MetaValue types.
* Change `ImportDoc.name: Option[String]` to `namespace: String`. If a name is not specified, the inferred name (the filename minus the .wdl extension) is used.
* Improve the "check" command to collect all type-checking errors and pretty-print them to stdout or a file.
* General clean-up of CLI, including switching from base classes to traits for handling common subcommand options.

### CODE REVIEW

File: src/main/scala/wdlTools/cli/package.scala [hunk 1]
```diff
@@ -369,25 +409,4 @@ class WdlToolsConf(args: Seq[String]) extends ScallopConf(args) {
   addSubcommand(readmes)
 
   verify()
-
-  /**
-    * The verbosity level
-    * @return
-    */
-  def verbosity: Verbosity = {
-    if (this.verbose.getOrElse(default = false)) {
```
Comment (orodeh) [inline src/main/scala/wdlTools/cli/package.scala:378]: What does the syntax `default = false` mean here? I have never seen it before.

File: src/main/scala/wdlTools/types/Unification.scala [hunk 2]
```diff
@@ -188,6 +184,7 @@ case class Unification(conf: Options) {
   //    T_Var(0) -> T_Int
   //    T_Var(1) -> T_String
   //
+  @throws(classOf[TypeUnificationException])
```
Comment (orodeh) [inline src/main/scala/wdlTools/types/Unification.scala:187]: If you want to use this kind of decoration please add a comment in a central location that we don't use it throughout the code in a rigorous manner. This is important when someone reads a method that *does not* have a `throws` decoration. It means that it *may* throw an exception, the omission is not a guarantee that no exception will be thrown.

File: src/main/scala/wdlTools/cli/package.scala (hunk 1 above)
Comment (jdidion) [reply src/main/scala/wdlTools/cli/package.scala:378]: Default is just the parameter name. It is not required, but my IDE suggests adding parameter names for boolean parameters to make the code a bit easier to read. For something like getOrElse I guess it's not really necessary.

File: src/main/scala/wdlTools/types/Unification.scala (hunk 2 above)
Comment (jdidion) [reply src/main/scala/wdlTools/types/Unification.scala:187]: oops - left that there by mistake

**Pull request merged**